src/sampling.jl

```diff
@@ -109,17 +116,29 @@ end
 knuths_sample!(a::AbstractArray, x::AbstractArray; initshuffle::Bool=true) =
     knuths_sample!(Base.GLOBAL_RNG, a, x; initshuffle=initshuffle)
 
-# Fisher-Yates sampling
-#
-#   create an array of index inds = [1:n]
-#
-#   for i = 1:k
-#       swap inds[i] with a random one in inds[i:n]
-#       set x[i] = a[inds[i]]
-#   end
-#
-#   O(n) for initialization + O(k) for random shuffling
-#
+"""
+    fisher_yates_sample!([rng], a::AbstractArray, x::AbstractArray)
+
+Fisher-Yates shuffling (with early termination).
+
+Pseudo-code:
+```
+n = length(a)
+k = length(x)
+
+create an array of index inds = [1:n]
+
+for i = 1:k
+    swap inds[i] with a random one in inds[i:n]
```

Comment:
I tried that, but the correct code is actually more complex than what you posted, and in the end it was almost identical to the source, so I went back to pseudo-code. People can easily look at the source if they want details.

(BTW, that function probably doesn't work for a long time given that it calls `randi`, which doesn't exist.)